egg/core/interaction.py

```diff
@@ -2,74 +2,32 @@
 
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
-
+import functools
+import warnings
 from dataclasses import dataclass
-from typing import Dict, Iterable, Optional
+from typing import Dict, Iterable, Optional, Union
 
 import torch
 import torch.distributed as distrib
 
 from egg.core.batch import Batch
 
 
-@dataclass(repr=True, eq=True)
-class LoggingStrategy:
-    store_sender_input: bool = True
-    store_receiver_input: bool = True
-    store_labels: bool = True
-    store_aux_input: bool = True
-    store_message: bool = True
-    store_receiver_output: bool = True
-    store_message_length: bool = True
-
-    def filtered_interaction(
-        self,
-        sender_input: Optional[torch.Tensor],
-        receiver_input: Optional[torch.Tensor],
-        labels: Optional[torch.Tensor],
-        aux_input: Optional[Dict[str, torch.Tensor]],
-        message: Optional[torch.Tensor],
-        receiver_output: Optional[torch.Tensor],
-        message_length: Optional[torch.Tensor],
-        aux: Dict[str, torch.Tensor],
-    ):
-
-        return Interaction(
-            sender_input=sender_input if self.store_sender_input else None,
-            receiver_input=receiver_input if self.store_receiver_input else None,
-            labels=labels if self.store_labels else None,
-            aux_input=aux_input if self.store_aux_input else None,
-            message=message if self.store_message else None,
-            receiver_output=receiver_output if self.store_receiver_output else None,
-            message_length=message_length if self.store_message_length else None,
-            aux=aux,
-        )
-
-    @classmethod
-    def minimal(cls):
-        args = [False] * 7
-        return cls(*args)
-
-    @classmethod
-    def maximal(cls):
-        return cls()
-
-
 @dataclass(repr=True, unsafe_hash=True)
 class Interaction:
     # incoming data
-    sender_input: Optional[torch.Tensor]
-    receiver_input: Optional[torch.Tensor]
-    labels: Optional[torch.Tensor]
-    aux_input: Optional[Dict[str, torch.Tensor]]
+    sender_input: Optional[Union[torch.Tensor, list]]
+    receiver_input: Optional[Union[torch.Tensor, list]]
+    labels: Optional[Union[torch.Tensor, list]]
+    aux_input: Optional[Dict[str, Union[torch.Tensor, list]]]
 
     # what agents produce
-    message: Optional[torch.Tensor]
-    receiver_output: Optional[torch.Tensor]
+    message: Optional[Union[torch.Tensor, list]]
+    receiver_output: Optional[Union[torch.Tensor, list]]
 
     # auxilary info
-    message_length: Optional[torch.Tensor]
-    aux: Dict[str, torch.Tensor]
+    message_length: Optional[Union[torch.Tensor, list]]
+    aux: Dict[str, Union[torch.Tensor, list]]
```

Comment:
EGG supports python 3.6 or newer, typing list objects with primitives instead of List has been introduced in a newer version. I'd keep the old one